x-pack/plugins/osquery/server/lib/fleet_integration.ts

```diff
@@ -34,11 +34,20 @@ export const getPackagePolicyDeleteCallback =
           await Promise.all(
             map(
               foundPacks.saved_objects,
-              (pack: { id: string; references: SavedObjectReference[] }) =>
+              (pack: {
+                id: string;
+                references: SavedObjectReference[];
+                attributes: { shards: Array<{ key: string; value: string }> };
+              }) =>
                 packsClient.update(
                   packSavedObjectType,
                   pack.id,
-                  {},
+                  {
+                    shards: filter(
```

Comment:
what is the purpose of this?

Reply:
Thanks @dasansol92  that's a great question 👍 
So in general: we have this connection between policies and packs. Where packs have reference to policies and policies to packs. Shards is a functionality of osquery that enables osquery to be run just on a % of agents in the policy. 
Now what this line does is to make sure we remove the reference to a policy from pack's shard, when the policy is deleted, so we do not have an EMPTY shard field in Pack. 

It fixes the behaviour shown in ticket 6238 (no link this time;p)